docs/source/basics/chat_datasets.rst

```diff
@@ -0,0 +1,192 @@
+    print(p_tokenizer.decode(tokens))
+    # '\nQ1 \n \nA1 \n'
+
+As an example, you can see the schema of the `SlimOrca dataset <https://huggingface.co/datasets/Open-Orca/SlimOrca-Dedup>`_.
+
+Specifying conversation style
+-----------------------------
+
+The structure of the conversation in the raw dataset can vary widely with different role names, different fields indicating
+where the message content name, and other ways. There are a few standardized formats that are common across many datasets.
+We have built-in converters to convert these standardized formats into a list of torchtune :class:`~torchtune.data.Message`
+that follows this format:
+
+.. code-block:: python
+
+    [
+        {
+            "role": "system" | "user" | "assistant" | "ipython",
+            "content": <message>,
+        },
+        ...
+    ]
+
+``"sharegpt"``
+^^^^^^^^^^^^^^
+The associated message transform is :class:`~torchtune.data.ShareGPTToMessages`. The expected format is:
+
+.. code-block:: python
+
+    {
+        "conversations": [
+            {
+                "from": "system" | "human" | "gpt",
+                "value": <message>,
+            },
+            ...
+        ]
+    }
+
+``"json"``
+^^^^^^^^^^
+The associated message transform is :class:`~torchtune.data.JSONToMessages`. The expected format is:
+
+.. code-block:: python
+
+    {
+        "messages": [
+            {
+                "role": "system" | "user" | "assistant",
+                "content": <message>,
+            },
+            ...
+        ]
+    }
+
+If your dataset does not fit one of the above conversation styles, then you will need to create a custom message transform.
+
+Loading chat datasets from Hugging Face
+---------------------------------------
+
+You need to pass in the dataset repo name to ``source``, select one of the conversation styles in ``conversation_style``, and specify the ``conversation_column``.
+For most HF datasets, you will also need to specify the ``split``.
+
+.. code-block:: python
+
+    from torchtune.models.gemma import gemma_tokenizer
+    from torchtune.datasets import chat_dataset
+
+    g_tokenizer = gemma_tokenizer("/tmp/gemma-7b/tokenizer.model")
+    ds = chat_dataset(
+        tokenizer=g_tokenizer,
+        source="Open-Orca/SlimOrca-Dedup",
+        conversation_column="conversations",
+        conversation_style="sharegpt",
+        split="train",
+    )
+
+.. code-block:: yaml
+
+    # Tokenizer is passed into the dataset in the recipe
+    dataset:
+      _component_: torchtune.datasets.chat_dataset
+      source: Open-Orca/SlimOrca-Dedup
+      conversation_column: conversations
+      conversation_style: sharegpt
+      split: train
+
+
+Loading local and remote chat datasets
+--------------------------------------
+
+To load in a local or remote dataset via https that has conversational data, you need to additionally specify the ``data_files`` and ``split``
```

Comment:
Can you add code here showing how to create a dummy "data/my_data.json"? I dont know where "split" should go

Also not sure what it should like when its a .csv or .txt